compiler-plugin/src/test/kotlin/arrow/meta/quotes/transform/TransformNewSourceTest.kt

```diff
@@ -0,0 +1,88 @@
+package arrow.meta.quotes.transform
+
+import arrow.meta.plugin.testing.CompilerTest
+import arrow.meta.plugin.testing.assertThis
+import arrow.meta.quotes.transform.plugins.TransformMetaPlugin
+import org.junit.Test
+
+class TransformNewSourceTest {
+  
+  @Test
+  fun `validate single extra file is created`() {
+    assertThis(CompilerTest(
+      config = { metaDependencies + addMetaPlugins(TransformMetaPlugin()) },
+      code = {
+        """ class NewSource {} """.source
+      },
+      assert = { quoteFileMatches("NewSource_Generated.kt",
+        """
+           package arrow
+           class NewSource_Generated {
+            fun sayHi() = println("Hi!")
+           }
+        """.source
+      )}
```

Comment:
I think it would be interesting to add an expression evaluation on the generated classes for all the tests.

Reply:
And if expression evaluation is added, it's not necessary to check the content of generated code (`quoteFileMatches`).